repl.fnl

```diff
@@ -0,0 +1,45 @@
+(local coroutine (require :coroutine))
+(local fennel (require :fennel))
+(local jeejah (require :jeejah))
+(local view (require :fennelview))
+
+(fn fennel-middleware
+  [f msg]
+  (match msg.op
+    "load-file" (let [f (assert (io.open msg.filename "rb"))]
+                  (tset msg
+                        :op "eval"
+                        :code (-> f
+                                  (: :read "*all")
+                                  (: :gsub "^#![^\n]*\n" "")))
+                  (: f :close))
+    (f msg)))
+
+(local repl-coro-freq 0.05)
+
+(fn run
+  [server]
+  (let [repl-coro server
+        repl-spin (fn [] (coroutine.resume repl-coro))
+        repl-chk (fn [] (not= (coroutine.status repl-coro) "dead"))]
+    (hs.timer.doWhile repl-chk repl-spin repl-coro-freq)))
+
+(fn start-opts
```

Comment:
Do we need a separate start and start-opts  function?

Reply:
I wasn't sure how/if multiple arities worked